tests/cnf/ran/gitopsztp/tests/ztp-bios-day-zero.go

```diff
@@ -0,0 +1,127 @@
+	. "github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/raninittools"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/version"
+	"github.com/openshift-kni/eco-gotests/tests/internal/cluster"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("ZTP BIOS Configuration Tests", Label(tsparams.LabelBiosDayZeroTests), func() {
+	var (
+		spokeClusterName string
+		nodeNames        []string
+	)
+
+	// 75196 - Check if spoke has required BIOS setting values applied
+	It("Check if SNO spoke has required BIOS setting values applied", reportxml.ID("75196"), func() {
+		versionInRange, err := version.IsVersionStringInRange(RANConfig.ZTPVersion, "4.17", "")
+		Expect(err).ToNot(HaveOccurred(), "Failed to check if ZTP version is in range")
+
+		if !versionInRange {
+			Skip("ZTP BIOS configuration tests require ZTP version of least 4.17")
+		}
+
+		spokeClusterName, err = GetSpokeClusterName(HubAPIClient, Spoke1APIClient)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get SNO cluster name")
+		By(fmt.Sprintf("Cluster name: %s", spokeClusterName))
+
+		nodeNames, err = GetNodeNames(Spoke1APIClient)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get node names")
+		By(fmt.Sprintf("Node names: %v", nodeNames))
+
+		By(fmt.Sprintf("cluster=%s SNO spoke=%s", spokeClusterName, nodeNames[0]))
+		hfs, err := bmh.PullHFS(HubAPIClient, nodeNames[0], spokeClusterName)
+		Expect(err).ToNot(
+			HaveOccurred(),
+			fmt.Sprintf("Failed to get HFS for spoke %s in cluster %s", nodeNames[0], spokeClusterName),
+		)
+
+		hfsObject, err := hfs.Get()
+		Expect(err).ToNot(
+			HaveOccurred(),
+			fmt.Sprintf("Failed to get HFS Obj for spoke %s in cluster %s", nodeNames[0], spokeClusterName),
+		)
+
+		hfsRequstedSettings := hfsObject.Spec.Settings
+		hfsCurrentSettings := hfsObject.Status.Settings
+		allSettingsMatch := true
+		if len(hfsRequstedSettings) > 0 {
+			Expect(len(hfsCurrentSettings) > 0).To(
+				BeTrueBecause("hfs.spec.settings map is not empty, but hfs.status.settings map is empty"))
+
+			for param, value := range hfsRequstedSettings {
+				setting, ok := hfsCurrentSettings[param]
+				if ok {
+					requestedSetting := value.String()
+					if requestedSetting == setting {
+						By(fmt.Sprintf("Requested setting matches current: %s=%s", param, setting))
+					} else {
+						By(
+							fmt.Sprintf("Requested setting %s value %s does not match current value %s", param, requestedSetting, setting))
+						allSettingsMatch = false
+					}
+				} else {
+					By(fmt.Sprintf("Current settings does not have param %s", param))
+				}
+			}
+
+			Expect(allSettingsMatch).To(BeTrueBecause("One or more requested settings does not match current settings"))
+		} else {
+			Skip("hfs.spec.settings map is empty")
+		}
+	})
+
+})
+
+// GetSpokeClusterName gets the spoke cluster name as string.
+func GetSpokeClusterName(hubAPIClient, spokeAPIClient *clients.Settings) (string, error) {
```

Comment:
it's gotten using a [different method](https://github.com/openshift-kni/eco-gotests/blob/main/tests/cnf/ran/internal/version/version.go#L92), but is [RANConfig.Spoke1Name](https://github.com/openshift-kni/eco-gotests/blob/main/tests/cnf/ran/internal/ranconfig/config.go#L56) useful here?

Reply:
```
ranconfig.Spoke1Config.Spoke1Name, err = version.GetClusterName(spoke1Kubeconfig)
```

naming of value vs function is confusing here.

Also  I don't think spokename == clustername is reliable, which is why I went with getting the cluster name programmatically. I copied that function from [another test suite](https://github.com/dgonyier/eco-gotests/blob/ztp-verify-bios/tests/assisted/ztp/internal/find/find.go#L27) in eco-gotests

I think there is one weird case for the kni-qe-66 ztp profile:

```
clustername = kni-qe-66
spokename = sno.kni-qe-66.lab.eng.rdu2.redhat.com
```

Reply:
@dgonyier Yes the version.GetClusterName just parses from the url in the kubeconfig so it looks like it may not work here. I'm curious how it's worked so far then since we only use ranconfig.Spoke1Name in the tests so far